.ci/chatops-commands/ci-problems-analyze.sh

```diff
@@ -0,0 +1,86 @@
+  >&2 echo "Checking workflow run $workflow_run_id attempt $workflow_run_attempt for problems..."
+
+  gh api -X GET "repos/$OWNER_NAME/$REPO_NAME/actions/runs/$workflow_run_id/attempts/$workflow_run_attempt/jobs?per_page=100" --jq '.jobs[] | select(.conclusion=="failure" and .name!="check-results" and .name!="zeebe-ci / Test summary") .id' | while read -r job_id; do
+    >&2 echo "Checking failed job $job_id for well known failure reasons..."
+
+    workflow_name=$(gh api -X GET "repos/$OWNER_NAME/$REPO_NAME/actions/jobs/$job_id" --jq '.workflow_name')
+    job_url="https://github.com/$OWNER_NAME/$REPO_NAME/actions/runs/$workflow_run_id/job/$job_id?pr=$PR_NUMBER"
+    job_annotations=$(gh api -X GET "/repos/$OWNER_NAME/$REPO_NAME/check-runs/$job_id/annotations")
+
+    ############################################################################
+    runner_problem_annotations=$(echo "$job_annotations" | jq '.[] | select(.message | contains("lost communication with the server") or contains("runner has received a shutdown signal"))')
+    if echo "$runner_problem_annotations" | grep -q message; then
+      # get runner name from job properties (usually empty on aborts) or from annotations via regex
+      runner_name=$(gh api "repos/$OWNER_NAME/$REPO_NAME/actions/jobs/$job_id" --jq '.runner_name' | tr '[:upper:]' '[:lower:]')
+      if [ "$runner_name" == "" ]; then
+        runner_name=$(echo "$runner_problem_annotations" | grep -oP '(?<=The self-hosted runner: )\S+')
+      fi
+
+      >&2 echo "Job $job_id got aborted due to problem with runner: $runner_name"
+
+      pods_dashboard_url="https://dashboard.int.camunda.com/d/000000019/pods?orgId=1&var-namespace=camunda&var-pod=$runner_name&var-container=All&from=now-7d&to=now"
+      gke_logs_url="https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_cluster%22%0Aresource.labels.project_id%3D%22ci-30-162810%22%0Aresource.labels.location%3D%22europe-west1%22%0Aresource.labels.cluster_name%3D%22camunda-ci%22%20$runner_name;duration=P7D?project=ci-30-162810"
+
+      echo "* 👻 [Job $job_id]($job_url) got aborted due to problems with runner \`$runner_name\`:"
+      echo "  * GitHub prematurely lost connection to the runner which can happen due to high job CPU load (try reducing load), network issues or hardware failures (try rerunning)."
+      echo "  * Check [resource usage for runner \`$runner_name\`]($pods_dashboard_url) and [Kubernetes logs for runner \`$runner_name\`]($gke_logs_url)."
+    fi
+
+    ############################################################################
+    timeout_annotations=$(echo "$job_annotations" | jq '.[] | select(.message | contains("has exceeded the maximum execution time"))')
+    if echo "$timeout_annotations" | grep -q message; then
+      runner_name=$(gh api "repos/$OWNER_NAME/$REPO_NAME/actions/jobs/$job_id" --jq '.runner_name' | tr '[:upper:]' '[:lower:]')
+
+      >&2 echo "Job $job_id got cancelled due to timeout."
```

Comment:
Do you think it is a good idea to print also the workflow name for better UX?

Reply:
This line specifically prints to stderr and is only for debugging where the ID is needed for unique identification (copy&paste) but your remark in general is right! I adjusted the end-user-facing parts of the script to output `<workflow_name> / <job_name>` for better readability.